src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/PublicationControllerTests.cs

```diff
@@ -1,114 +1,745 @@
+            };
+
+            var response = await CreatePublication(client, request);
+
+            var validationProblem = response.AssertValidationProblem();
+
+            validationProblem.AssertHasGlobalError(ValidationErrorMessages.PublicationSlugNotUnique);
+
+            await using var context = TestApp.GetDbContext<ContentDbContext>();
 
-        Assert.IsType<PublicationCreateViewModel>(result.Value);
-        Assert.Equal(topicId, result.Value.Topic.Id);
+            // Assert a new publication was not created
+            var saved = Assert.Single(await context.Publications.ToListAsync());
+            Assert.Equal(publication.Id, saved.Id);
+        }
+
+        private static async Task<HttpResponseMessage> CreatePublication(HttpClient client,
+            PublicationCreateRequest request)
+        {
+            return await client.PostAsJsonAsync("api/publications", request);
+        }
     }
 
-    [Fact]
-    public async Task CreatePublication_ValidationFailure()
+    public class ListPublicationDataSetsTests(TestApplicationFactory testApp) : PublicationControllerTests(testApp)
     {
-        var publicationService = new Mock<IPublicationService>();
+        [Theory]
+        [InlineData(1, 10, 2)]
+        [InlineData(1, 2, 10)]
+        [InlineData(2, 2, 10)]
+        public async Task PublicationHasMultipleDataSets_Success(
+            int page,
+            int pageSize,
+            int numberOfAvailableDataSets)
+        {
+            Publication publication = DataFixture
+                .DefaultPublication();
+
+            var dataSets = DataFixture
+                .DefaultDataSet()
+                .WithStatusPublished()
+                .WithPublicationId(publication.Id)
+                .GenerateList(numberOfAvailableDataSets);
+
+            await TestApp.AddTestData<ContentDbContext>(context => context.Publications.Add(publication));
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.AddRange(dataSets));
+
+            var dataSetVersions = dataSets
+                .Select(ds =>
+                    DataFixture
+                        .DefaultDataSetVersion(
+                            filters: 1,
+                            indicators: 1,
+                            locations: 1,
+                            timePeriods: 2)
+                        .WithStatusPublished()
+                        .WithDataSet(ds)
+                        .FinishWith(dsv => ds.LatestLiveVersion = dsv)
+                        .Generate())
+                .ToList();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.AddRange(dataSetVersions);
+                context.DataSets.UpdateRange(dataSets);
+            });
+
+            var expectedDataSetIds = dataSets
+                .OrderByDescending(ds => ds.LatestLiveVersion != null ? ds.LatestLiveVersion.Published : DateTimeOffset.MinValue)
```

Comment:
Yes it can be, I only did it like that so it would work regardless of the test data set up.

I've simplified it now from 

```csharp
var expectedDataSetIds = dataSets
    .OrderByDescending(ds =>
        ds.LatestLiveVersion != null ? ds.LatestLiveVersion.Published : DateTimeOffset.MinValue)
    .ThenBy(ds => ds.Title)
    .ThenBy(ds => ds.Id)
    .Select(ds => ds.Id)
    .Skip((page - 1) * pageSize)
    .Take(pageSize)
    .ToList();
```

to

```csharp
var expectedDataSetIds = dataSets
    .OrderByDescending(ds => ds.LatestLiveVersion!.Published)
    .Select(ds => ds.Id)
    .Skip((page - 1) * pageSize)
    .Take(pageSize)
    .ToList();
```